tests/test_tools.py

```diff
@@ -195,6 +197,121 @@ def test_set_output_directory_bad(user_filesystem):
         assert not Path(actual_args.output_directory).is_dir()
 
 
+@pytest.mark.parametrize(
+    "inputs, expected",
+    [
+        # Test when only a home config file exists (no local config file),
+        # expect to return args if wavelength or anode type is specified,
+        # otherwise update args with values from the home config file.
+        # C1: no args, expect to update arg values from home config
+        ([""], {"wavelength": 0.3, "anode_type": None}),
+        # C2: wavelength provided, expect to return args unchanged
+        (["--wavelength", "0.25"], {"wavelength": 0.25, "anode_type": None}),
+        # C3: anode type provided, expect to return args unchanged
+        (["--anode-type", "Mo"], {"wavelength": None, "anode_type": "Mo"}),
+        # C4: both wavelength and anode type provided,
+        # expect to return args unchanged
+        (
+            ["--wavelength", "0.7", "--anode-type", "Mo"],
+            {"wavelength": 0.7, "anode_type": "Mo"},
+        ),
+    ],
+)
+def test_load_wavelength_from_config_file_with_home_conf_file(
+    mocker, user_filesystem, inputs, expected
+):
+    cwd = Path(user_filesystem)
+    home_dir = cwd / "home_dir"
+    mocker.patch("pathlib.Path.home", lambda _: home_dir)
+    os.chdir(cwd)
+
+    cli_inputs = ["2.5", "data.xy"] + inputs
+    actual_args = get_args(cli_inputs)
+    actual_args = _load_wavelength_from_config_file(actual_args)
+    assert actual_args.wavelength == expected["wavelength"]
+    assert actual_args.anode_type == expected["anode_type"]
+
+
+@pytest.mark.parametrize(
+    "inputs, expected",
+    [
+        # Test when a local config file exists,
+        # expect to return args if wavelength or anode type is specified,
+        # otherwise update args with values from the home config file.
+        # Results should be the same whether if the home config exists.
+        # C1: no args, expect to update arg values from local config
+        ([""], {"wavelength": 0.6, "anode_type": None}),
+        # C2: wavelength provided, expect to return args unchanged
+        (["--wavelength", "0.25"], {"wavelength": 0.25, "anode_type": None}),
+        # C3: anode type provided, expect to return args unchanged
+        (["--anode-type", "Mo"], {"wavelength": None, "anode_type": "Mo"}),
+        # C4: both wavelength and anode type provided,
+        # expect to return args unchanged
```

Comment:
I agree that the tests should be the same here, as they are, but we then need to modify the last test in both places.

Also maybe mention in the preamble that the home config file has wavelength of 0.3 and the local config has wavelength of 0.6 to make the test easier to read.

A thought occurred to me that maybe having both wavelength and anode_type specified is tested elsewhere and you do want to return both values here but have it fail in a test elsewhere, but i fthat is the case, say it explicitly in the comment about the test.